### PR TITLE
Fix coverage of unit tests

### PR DESCRIPTION
**Pull request by ilaner**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/lint/linter.py [hunk 1]
```diff
@@ -816,7 +815,8 @@ def _docker_run_pytest(self, test_image: str, keep_container: bool, test_xml: st
                     with open(file=xml_apth, mode='bw') as f:
                         f.write(test_data_xml)  # type: ignore
 
-                if cov:
+                if not no_coverage:
```
Comment (dorschw) [inline demisto_sdk/commands/lint/linter.py:818]: why not `if cov`? `if not` is a tiny bit harder to read imo

File: demisto_sdk/commands/lint/linter.py [hunk 2]
```diff
@@ -782,8 +782,7 @@ def _docker_run_pytest(self, test_image: str, keep_container: bool, test_xml: st
         test_json = {}
         try:
             # Running pytest container
-            cov_file_path = Path.joinpath(self._pack_abs_dir, '.coverage')
-            cov = self._pack_abs_dir.stem if not no_coverage and cov_file_path.exists() else ''
```
Comment (dorschw) [inline demisto_sdk/commands/lint/linter.py:786]: what happens now if `cov_file_path` doesn't exist?

File: demisto_sdk/commands/lint/linter.py [hunk 3]
```diff
@@ -816,7 +815,8 @@ def _docker_run_pytest(self, test_image: str, keep_container: bool, test_xml: st
                     with open(file=xml_apth, mode='bw') as f:
                         f.write(test_data_xml)  # type: ignore
 
-                if cov:
+                if not no_coverage:
+                    cov_file_path = os.path.join(self._pack_abs_dir, '.coverage')
```
Comment (dorschw) [inline demisto_sdk/commands/lint/linter.py]: 1. please use Pathlib
2. please make sure the file exists and handle when it doesn't

File: demisto_sdk/commands/lint/linter.py (hunk 1 above)
Comment (ilaner) [reply demisto_sdk/commands/lint/linter.py:819]: That's what I thought as well, but there is a reason for this. `if cov` tells us if the file exists. But if it doesn't and we didn't specify for `no-coverage` we would like to create it.

File: demisto_sdk/commands/lint/linter.py [hunk 4]
```diff
@@ -783,7 +783,7 @@ def _docker_run_pytest(self, test_image: str, keep_container: bool, test_xml: st
         try:
             # Running pytest container
             cov_file_path = Path.joinpath(self._pack_abs_dir, '.coverage')
-            cov = self._pack_abs_dir.stem if not no_coverage and cov_file_path.exists() else ''
+            cov = self._pack_abs_dir.stem if not no_coverage else ''
```
Comment (dorschw) [inline demisto_sdk/commands/lint/linter.py:786]: i'd reverse it, (`'' if no coverage else ...`), but it's a matter of taste.

**Review by dorschw — APPROVED** [review]

File: demisto_sdk/commands/lint/linter.py (hunk 4 above)
Comment (dorschw) [reply demisto_sdk/commands/lint/linter.py:786]: also, do we intentionally remove the `and cov_file_path.exists()`?

File: demisto_sdk/commands/lint/linter.py (hunk 1 above)
Comment (dorschw) [reply demisto_sdk/commands/lint/linter.py]: On one hand, I'd assume there may have been a reason for that in the past (is the author/maintainer still available to answer?) - does removing it really save us a significant amount of time/calls?

If it does, let's hear what @tkatzir says

File: demisto_sdk/commands/lint/linter.py (hunk 4 above)
Comment (ilaner) [reply demisto_sdk/commands/lint/linter.py:786]: yes, pytest will create this file

**Pull request merged**